testing/e2e/clients/dns.go

```diff
@@ -0,0 +1,207 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/logger"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/dns/armdns"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/privatedns/armprivatedns"
+)
+
+type zone struct {
+	name           string
+	subscriptionId string
+	resourceGroup  string
+}
+
+type privateZone struct {
+	name           string
+	subscriptionId string
+	resourceGroup  string
+}
+
+// ZoneOpt specifies what kind of zone to create
+type ZoneOpt func(z *armdns.Zone) error
+
+// PrivateZoneOpt specifies what kind of private zone to create
+type PrivateZoneOpt func(z *armprivatedns.PrivateZone) error
+
+func NewZone(ctx context.Context, subscriptionId, resourceGroup, name string, zoneOpts ...ZoneOpt) (*zone, error) {
+	name = nonAlphanumericRegex.ReplaceAllString(name, "")
+	name = name + ".com"
```

Comment:
is adding `.com` a convention the caller won't know?

Reply:
yes although I believe this still works if it's test.com.com haha